src/binary/uint.rs

```diff
@@ -132,30 +168,34 @@ mod tests {
         let data = &[0b1000_0000];
         let uint = DecodedUInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
         assert_eq!(uint.size_in_bytes(), 1);
-        assert_eq!(uint.value(), 128);
+        assert_eq!(uint.value(), &UInteger::U64(128));
     }
 
     #[test]
     fn test_read_two_byte_uint() {
         let data = &[0b0111_1111, 0b1111_1111];
         let uint = DecodedUInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
         assert_eq!(uint.size_in_bytes(), 2);
-        assert_eq!(uint.value(), 32_767);
+        assert_eq!(uint.value(), &UInteger::U64(32_767));
     }
 
     #[test]
     fn test_read_three_byte_uint() {
         let data = &[0b0011_1100, 0b1000_0111, 0b1000_0001];
         let uint = DecodedUInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
         assert_eq!(uint.size_in_bytes(), 3);
-        assert_eq!(uint.value(), 3_966_849);
+        assert_eq!(uint.value(), &UInteger::U64(3_966_849));
     }
 
     #[test]
-    fn test_read_uint_overflow() {
-        let data = &[0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF, 0x01];
+    fn test_read_uint_too_large() {
+        let mut buffer = Vec::with_capacity(MAX_UINT_SIZE_IN_BYTES + 1);
```

Comment:
PR tour: now that we can read an integer of (practically) any size, this test had to be rewritten to pass a `UInt` buffer of 2049 bytes, one byte in length beyond the configured maximum allowable.